### ISSUE TITLE
dependsOn projects exclusion does not work

### ISSUE THREAD

**Issue by ixahmedxi** [nrwl/nx#26345]
### Current Behavior

First of all a bit of context, I am trying to move [orbitkit](https://orbitkit.dev) to nx, it is a fully custom monorepo, so not using integrated monorepo architecture.

I have a `packages/config` directory that hosts a few packages responsible for sharing tools config preset to the workspace, for example a `storybook` and a `vite` package.

I then have a package called `ui` in `packages/ui` that is a UI components library with storybook, and is bundled up using `tsup`.
The flow is the following:
- To run storybook on the ui library, there is a dedicated `storybook` script.
- To build storybook, it's `build-storybook`.
- To build the library itself in watch mode, I use the `dev` command, which runs `tsup --watch`.

For storybook's functionality in the ui package to work, I do need to have the `storybook` and `vite` packages built, however that is not a requirement for the normal `dev` and `build` commands of the package itself. I thought then about a way where I can exclude the dependency over the `storybook` and `vite` config packages from the `dev` and `build` tasks in the `dependsOn` configuration like so:
![image](https://github.com/nrwl/nx/assets/20271968/cf7f17e9-2423-4003-b743-10cf8d9e36d6)

This was suggested to me by NX console, so I thought it is correct, however when running it I get the following error:

![image](https://github.com/nrwl/nx/assets/20271968/b89d7203-89b6-4b62-ac21-eae02a2c0da1)

### Expected Behavior

I expect the `!vite` kind of configuration in the `dependsOn` to exclude this package from being needed to be built beforehand.

### GitHub Repo

https://github.com/ixahmedxi/orbitkit/tree/nx

### Steps to Reproduce

1. Clone the Github repo above
2. Run `bun install`
3. Run `bun nx dev ui`

### Nx Report

```shell
Node   : 20.12.2
OS     : linux-x64
bun    : 1.1.10

nx          : 19.1.2
@nrwl/tao   : 19.1.2
typescript  : 5.4.5
```

### Package Manager Version

Bun 1.1.10

### Operating System

- [X] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by JamesHenry** [comment]
Hi @ixahmedxi, if you want to only negate projects from consideration in a project selector, right now you just need to reference `*` in the array as well.

i.e.

```json
"projects": ["*", "!vite", "!storybook"]
```

We did briefly add support for including only negative patterns, but it was reverted here seemingly to fix an unintended/undesired side-effect: https://github.com/nrwl/nx/pull/23459

@AgentEnder is it being tracked to add that back again?

**Comment by JamesHenry** [comment]
@ixahmedxi I have deleted my previous incorrect comment, I just checked and the entries in `projects` are currently processed one by one, so combining `*` with the negation to create an overall result does not work.

https://github.com/nrwl/nx/pull/23459 reverted the change that would have allowed this because it seems it caused some unwanted side-effects. 

@AgentEnder is it being tracked to add this back in?
